### DISCUSSION TITLE
programatically saving content type data in decouple cms

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11298] [Q&A]
Folks need ur help in below:

I created a content type subscriber on admin side:
 
![image](https://user-images.githubusercontent.com/5202291/156826791-737f5bda-ce85-4329-901b-64d59bec2cf3.png)

I wanna require to save newsletter subscriber email addresses programmtically:

so I used the below code but that code not save my data:

Saving Code:

```C#
private async Task<bool> CreatePOST(string id,string EmailAddress, string returnUrl, bool stayOnSamePage, Func<ContentItem, Task> conditionallyPublish)
        {
            var contentItem = await _contentManager.NewAsync(id);
             contentItem.Owner = "Admin";
            contentItem.DisplayText = EmailAddress;
            ContentItem contentItem1 = contentItem;
            var model = await _contentItemDisplayManager.UpdateEditorAsync(contentItem1, _updateModelAccessor.ModelUpdater, true);

            await _contentManager.CreateAsync(contentItem1, VersionOptions.Draft);

            await conditionallyPublish(contentItem1);

            if ((!string.IsNullOrEmpty(returnUrl)) && (!stayOnSamePage))
            {
                return true;
            }           
            return true;
        }
```

and i sending the data from ajax like below:

```javascript
$('#btnSubscribeNewsletter').click(function (event) {
                debugger;

                var subscriptionEmail = $('#SubscriptionEmail').val().trim();
                if (subscriptionEmail == "") {
                    alert('Please provide the e-mail address');
                    return;
                }
                var pattern = /^\b[A-Z0-9._%-]+@@[A-Z0-9.-]+\.[A-Z]{2,4}\b$/i

                if (!pattern.test(subscriptionEmail)) {
                    alert('Please provide the valid e-mail address');
                    return;
                }

                var email = {};
                email.SubscriptionEmail = subscriptionEmail;
                var token = $("input[name='__RequestVerificationToken']").val();

                $.ajax({
                    type: 'POST',
                    url: '/OrchardCore.Users/User/SavingNewsletter',
                    data: JSON.stringify(email),
                    contentType: "application/json; charset=utf-8",
                    dataType: "json",
                }).done(function (response) {
                    alert("Thanks for the subscription");
                }).fail(function (XMLHttpRequest, textStatus, errorThrown) {
                    alert(response.responseText);
                });


            });
```

Controller part:

```C#
[AllowAnonymous]
        [IgnoreAntiforgeryToken]
        [HttpPost]
        public async Task<ActionResult> SavingNewsletterAsync([FromBody] Email email)
        {
            var _savingNewsletter = new SavingNewsletter(_contentManager,_contentItemDisplayManager,_updateModelAccessor, _contentDefinitionManager);
            await CreateAndPublishPOST("Subscribers", "", "");
            if (email == null)
            {
                return View();
            }
            else
            {
                await _savingNewsletter.CreateNewsletter(email.SubscriptionEmail);
                return View();
            }
        }
```

Below is the screen shot of front pages:

Email Input front site like that:
![image](https://user-images.githubusercontent.com/5202291/156830109-6c0f9141-4e71-4024-be30-d91b9969dfd1.png)

![image](https://user-images.githubusercontent.com/5202291/156830502-cbb32079-e9b9-4587-96ef-ef6b76b21965.png)

![image](https://user-images.githubusercontent.com/5202291/156830531-077e8ef0-762b-418d-993a-f62e9137fc1e.png)

![image](https://user-images.githubusercontent.com/5202291/156830632-c655495a-3ffe-4c7d-981b-fb5bd0311f57.png)

so it not update my values also not inserting please advise folks:

![image](https://user-images.githubusercontent.com/5202291/156830778-d3146867-0f3d-44f9-a95e-b75f42c0b5df.png)

**Comment by Skrypt** [comment]
The best thing would be to copy what's in the AdminController of the OrchardCore.Contents module. It already works with any Content-Type. If there is something that doesn't work then it's related to what you push to these methods.

**Comment by saleemhaq** [comment]
I shared my whole code based can you please assist me how I achieved this
take my whole week but unable to achieved still facing trouble.

I copy as it is from admincontroller.cs and I hardcoded content type but
there values not set when it going to be update?

Please advise.

**Comment by Skrypt** [comment]
What you are trying to do is to change the Owner and Author of the ContentItem?

**Comment by saleemhaq** [comment]
Just want to save data of my content type Subscriber.

But it not inserting? Please advise

**Comment by ns8482e** [comment]
@saleemhaq there are two CRUD sample code available on `OrchardCore.Demo` module. Refer `ContentController` for CRUD of content item and `TodoController` for POCO

Check and see what you are missing in your code
